mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp

```diff
@@ -60,9 +60,51 @@ struct ConcatOptimization : public OpRewritePattern<tosa::ConcatOp> {
   }
 };
 
+struct SelfConcatToTile : public OpRewritePattern<tosa::ConcatOp> {
+  using OpRewritePattern<tosa::ConcatOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(tosa::ConcatOp op,
+                                PatternRewriter &rewriter) const override {
+    if (llvm::all_equal(op->getUsers())) {
+      const auto concatUser =
+          llvm::dyn_cast<tosa::ConcatOp>(op->getUses().begin()->getOwner());
```

Comment:
nit: 
``` suggestion
          dyn_cast<tosa::ConcatOp>(op->getUses().begin()->getOwner());
```

Reply:
The general style in the file seems to be to use llvm::dyn_cast